src/main/kotlin/com/cognifide/gradle/aem/instance/content/Node.kt

```diff
@@ -0,0 +1,109 @@
+package com.cognifide.gradle.aem.instance.content
+
+import com.cognifide.gradle.aem.instance.InstanceSync
+import com.jayway.jsonpath.DocumentContext
+import com.jayway.jsonpath.JsonPath
+import net.minidev.json.JSONArray
+import org.apache.jackrabbit.vault.util.JcrConstants
+
+class Node private constructor() {
+
+    lateinit var sync: InstanceSync
+    lateinit var path: String
+    lateinit var name: String
+    lateinit var document: DocumentContext
+
+    fun children(): Iterator<Node> {
+        return sync.get("$path.harray.1.json") { asJson(it) }
+                .read<JSONArray>("__children__")
+                .map { props -> loadChild(sync, path, props as MutableMap<*, *>) }
+                .iterator()
+    }
+
+    fun get(propName: String): Any = document.read(propName) as Any
```

Comment:
What about:
`fun prop(name: String):any` and `fun propString(name: String):String`

Reply:
for me it is more a value than a prop

Reply:
avoid typing / adding dummy methods for casting. this will be responsibility of scripts so that methods returning `Any` or `Any?` are fine for me. eventualy one extra method for method

```kotlin
fun property(name: String): Any?
fun string(name: String): String? = property(name)?.toString()
```

but still such method does not give any important value